pycheribuild/projects/release.py

```diff
@@ -0,0 +1,162 @@
+
+class Tag(SimpleProject):
+    target = "tag"
+    
+    @classmethod
+    def setup_config_options(cls, **kwargs):
+        super().setup_config_options(**kwargs)
+
+        cls.gittag = cls.add_config_option("gittag", help="Tag to apply")
+
+    def __init__(self, config: CheriConfig):
+        super().__init__(config)
+
+    def process(self):
+        source_root = self.config.source_root
+        for repo in repos:
+            run_command("git", "-C", repo, "tag", self.gittag, cwd=source_root)
+
+class UnTag(SimpleProject):
+    target = "untag" 
+
+    @classmethod
+    def setup_config_options(cls, **kwargs):
+        super().setup_config_options(**kwargs)
+
+        cls.gittag = cls.add_config_option("gittag", help="Tag to apply")
+
+    def __init__(self, config: CheriConfig):
+        super().__init__(config)
+
+    def process(self):
+        source_root = self.config.source_root
+        for repo in repos:
+            run_command("git", "-C", repo, "tag", "-d", self.gittag, cwd=source_root)
+
+class Release(SimpleProject):
+    project_name = "release"
+    do_not_add_to_targets = True
+
+    def __init__(self, config: CheriConfig):
+        super().__init__(config)
+
+    def process(self):
+        source_root = self.config.source_root
+
+        Path(source_root, "cheribuild/cheribuild.json").write_text("""{
+            "source-root": "../../sources",
+            "build-root": "../../build",
+            "output-root": "../../output",
+            "skip-update": true
+} 
+            """)
+
+
+class MorelloRelease(Release):
+    target = "morello-release"
+    dependencies = ["morello-llvm", "cheribsd-morello-purecap",
+                    "gdb-native",
+                    "gdb-morello-hybrid-for-purecap-rootfs",
+                    "arm-none-eabi-toolchain", "morello-acpica",
+                    "morello-scp-firmware",
+                    "morello-trusted-firmware",
+                    "morello-flash-images",
+                    "disk-image-morello-purecap"]
+    def __init__(self, config: CheriConfig):
+        super().__init__(config)
+
+
+    def process(self):
+        super().process()
+
+        root = self.config.output_root.parent
+        output_root = self.config.output_root
+
+        install_script = Path(output_root, "install_and_run_fvp.sh")
+        install_script.write_text("""#!/bin/sh
+        dir=$(CDPATH= cd -- "$(dirname -- "$0")" && pwd)
+        exec "${dir}/cheribuild.py" install-morello-fvp run-fvp-morello-purecap "$@""")
+        install_script.chmod(0o755)
+
+        run_command("bsdtar", "-cavf", output_root / "release-morello.tar.xz", 
+            "-C", root,
+            "--options=xz:threads=" + str(default_make_jobs_count()),
+            "--options=compression-level=9",  # reduce size a bit more
+            "--exclude=*.git",
+            "output/morello-sdk/firmware",
+            "output/cheribsd-morello-purecap.img",
+            "output/sdk",
+            "sources/cheribuild",
```

Comment:
Only because you've configured your cheribuild to have a non-standard layout. If I were to run this it wouldn't work. If you want a specific layout you need to assemble that yourself in a staging directory.